examples/model_selection/plot_cost_sensitive_learning.py

```diff
@@ -500,20 +500,18 @@ def plot_roc_pr_curves(vanilla_model, tuned_model, *, title):
 # a gain of 2% of the amount of the transaction. However, accepting a fraudulent
 # transaction result in a loss of the amount of the transaction. As stated in [2]_, the
 # gain and loss related to refusals (of fraudulent and legitimate transactions) are not
-# trivial to define. Here, we define that a refusal of a legitimate transaction is
-# estimated to a loss of 5€ while the refusal of a fraudulent transaction is estimated
-# to a gain of 50€ and the amount of the transaction. Therefore, we define the
-# following function to compute the total benefit of a given decision:
+# trivial to define. Here, we define that a refusal of a legitimate transaction
+# is estimated to a loss of 5€ while the refusal of a fraudulent transaction is
+# estimated to a gain of 50€. Therefore, we define the following function to
+# compute the total benefit of a given decision:
 
 
 def business_metric(y_true, y_pred, amount):
     mask_true_positive = (y_true == 1) & (y_pred == 1)
     mask_true_negative = (y_true == 0) & (y_pred == 0)
     mask_false_positive = (y_true == 0) & (y_pred == 1)
     mask_false_negative = (y_true == 1) & (y_pred == 0)
-    fraudulent_refuse = (mask_true_positive.sum() * 50) + amount[
-        mask_true_positive
-    ].sum()
```

Comment:
Not sure why the amount here is removed, when I gave a talk on this, this was a nice point to make.

Reply:
Because you don't gain the amount when rejecting a fraudulent case :). Indeed, you are just not loosing it.

The amount itself only contribute when accepting the claim by taking a proportional amount.

Reply:
Yes, but we have the false negative counterpart. Removing this puts a massive pressure on making sure we have no false negatives, and will somewhat be okay with true positives being low, since the amounts are usually a lot larger than 50€.

Reply:
The updated PR puts our example cost matrix better in line with the one proposed in [Elkan 2001 paper on cost-sensitive learning](https://cseweb.ucsd.edu/~elkan/rescale.pdf) (the fixed costs and gains are not the same but the variable components are compatible). Here is an excerpt of the relevant paragraph (part of section 1.2):

![image](https://github.com/scikit-learn/scikit-learn/assets/89061/0e82f351-e862-42a9-92f2-bf7cf0cddd5d)

I don't understand why we would put a gain that is proportional to amount in the `fraudulent_refuse` case. If you catch a frauder, nobody will pay the bank the amount of the transaction the frauder would have otherwise stolen.

Reply:
Fair enough. Would be nice to have this chart in the example actually, makes things quite clear.

Reply:
I'll keep that in mind for a future iteration on this example.